### ISSUE TITLE
twisted.conch.checkers should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8225]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8225|
|Type|enhancement|
|Created|2016-03-03 13:26:25Z|
|Branch|https://github.com/twisted/twisted/tree/conch-checkers-py3-8225|

<details><summary>Searchable metadata</summary>

```
trac-id__8225 8225
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_conch_checkers_py3_8225 branches/conch-checkers-py3-8225
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1457011585374706 1457011585374706
changetime__1457073720927364 1457073720927364
version__None None
owner__hawkowl hawkowl
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [46889]) Branching to conch-checkers-py3-8225.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

[#7958](https://github.com/twisted/twisted/issues/7958) is probably required

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

I'm happy with this. Builders spun, will be green soon. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Hi,

Thanks for the patch. It looks good.

Happy to see py3 love for conch.

Major comments:

1. Is it ok to remove the test_shadowGetByNameWithoutEither ? I would say that on Windows we don't have spwd.... also AIX / HPUX and some other unixes might lack the spwd module.

Not sure if major or minor:

1. Instead of converting the username/password in requestAvatarId I would say that we should update the credential initialization to do the conversion at init time...or a bit closer to the entry layer.

Minor comments:

1. Why the private new members in compat? I would say to use the same names as six.

2. Please consider fixing the few twistedchecker errors.

Please check my comments and resubmit or merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46908]) Merge conch-checkers-py3-8225: Port twisted.conch.checkers to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8225](https://github.com/twisted/twisted/issues/8225)
